apis/AVInput/IAVInput.h

```diff
@@ -0,0 +1,301 @@
+
+            // @text getEdidVersion
+            // @brief Returns the EDID version
+            // @param id - in - The ID of the input device to get the EDID version for
+            // @param version - out - The EDID version
+            virtual Core::hresult GetEdidVersion(int id /* @in */, string& version /* @out */) = 0;
+
+            // @text setEdid2AllmSupport
+            // @brief Sets an HDMI ALLM bit in EDID
+            // @param id - in - The ID of the input device to set the ALLM bit for
+            // @param allm - in - The ALLM bit value to set
+            virtual Core::hresult SetEdid2AllmSupport(int id /* @in */, bool allm /* @in */) = 0;
+
+            // @text getEdid2AllmSupport
+            // @brief Returns the ALLM bit in EDID
+            // @param id - in - The ID of the input device to get the ALLM
+            // @param allm - out - The ALLM bit value
+            virtual Core::hresult GetEdid2AllmSupport(int id /* @in */, bool& allm /* @out */) = 0;
+
+            // @text setVRRSupport
+            // @brief Sets an HDMI VRR support bit in EDID
+            // @param id - in - The ID of the input device to set the VRR
+            // @param vrrSupport - in - The VRR support bit value to set
+            virtual Core::hresult SetVRRSupport(int id /* @in */, bool vrrSupport /* @in */) = 0;
+
+            // @text getVRRSupport
+            // @brief Returns the VRR support bit in EDID
+            // @param id - in - The ID of the input device to get the VRR
+            // @param vrrSupport - out - The VRR support bit value
+            virtual Core::hresult GetVRRSupport(int id /* @in */, bool& vrrSupport /* @out */) = 0;
+
+            // @text getHdmiVersion
+            // @brief Returns the HDMI compatibility version for the specified input device
+            // @param id - in - The ID of the input device to get the HDMI version
+            // @param hdmiVersion - out - The HDMI compatibility version
+            virtual Core::hresult GetHdmiVersion(int id /* @in */, string& hdmiVersion /* @out */) = 0;
+
+            // @text setAudioMixerLevels
+            // @brief Sets the mixer levels for the specified input device
+            // @param levels - in - The mixer levels to set
+            virtual Core::hresult SetAudioMixerLevels(Exchange::IAVInput::MixerLevels levels /* @in */) = 0;
+
+            // @text startInput
+            // @brief Starts the specified input device
+            // @param id - in - The ID of the input device to start
+            // @param type - in - The type of input device to start
+            // @param audioMix - in - Whether to enable audio mixing
+            // @param planeType - in - The video plane type to use
+            // @param topMostPlane - in - Whether to use the top-most video plane
+            // TODO: Documentation states ID is optional. If optional, are any specified defaults inserted into the JSON prior to calling
```

Comment:
Discussed usage of `OptionalType<T>` with Marcel, this is only available in Thunder 5.x. How should we handle optional params until then for COM-RPC? Also it seems like some API documentation specifes optional fields but in code are not handled as optional, is this a known issue?